src/averageAges.js

```diff
@@ -37,7 +42,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womansList = people.filter(human => human.sex === 'f'
+    && (!withChildren || people.find(child => child.mother === human.name)
+    )
+  );
+
+  const womansAge = womansList.map(woman => woman.died - woman.born
+  ).reduce((a, b) => a + b);
+
+  return womansAge / womansList.length;
```

Comment:
you can create a separate function so as not to duplicate code

Reply:
good idea